openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -124,7 +127,22 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             ),
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
-    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and \
+            not user_access['is_enrolled'] and not course.invitation_only:
+        CourseHomeMessages.register_info_message(
+            request,
+            Text(_(
+                u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
+            )).format(
+                open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
+                close_enroll_link=HTML('</button>')
+            ),
+            title=Text(_(u'Welcome to {course_display_name}')).format(
+                course_display_name=course.display_name
+            )
+        )
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and \
+            not user_access['is_enrolled'] and course.invitation_only and allow_anonymous:
```

Comment:
If the course is set to `invitation_only`, the user should not be able to enroll by themself (i.e. the message should not contain the Enroll now button). Just a message text mentioning that the user should be enrolled in the course to have full access, should be good enough. The idea is, if the course is `invitation_only`, the students will be added to the course by the staff/course creators. So what you can do here is:
```
 if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
        if not course.invitation_only:
                 Display the message containing Enroll button
        else:
                 Display message text "You should be enrolled in the course to fully access the contents"
```

Reply:
Ah, ok. I get it. I was getting confused as to what an enroll message is.